### PR TITLE
KS-AMIS-F-04: crypto/matrix: add an upperbound of size of a matrix

### PR DESCRIPTION
**Pull request by cychuang0924**
**Before submitting a pull request**, please make sure the following is done:

### Checklist

- [x] Fork [the repository](https://github.com/getamis/alice) and create your new branch from `master`.
- [x] Please mention the name of community in Pull Request title.
- [x] Git Commit Messages
    - [x] Use the present tense (Also in Pull Request title): "Add feature" not "Added feature"
    - [x] Use the imperative mood (Also in Pull Request title): "Move cursor to..." not "Moves cursor to..."
    - [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.

### CODE REVIEW

File: crypto/matrix/matrix.go [hunk 1]
```diff
@@ -528,7 +539,8 @@ func (m *Matrix) DeleteColumn(nLowerIndex, nUpperIndex uint64) (*Matrix, error)
 	if err != nil {
 		return nil, err
 	}
-	return transposeM.Transpose(), nil
```
Comment (markya0616) [inline crypto/matrix/matrix.go:531]: rm this change

File: crypto/matrix/matrix_test.go [hunk 2]
```diff
@@ -672,7 +680,6 @@ var _ = Describe("Matrix test", func() {
 				{big.NewInt(0)},
 			}),
 		)
-
```
Comment (markya0616) [inline crypto/matrix/matrix_test.go:675]: rm this change

File: crypto/matrix/matrix.go [hunk 3]
```diff
@@ -425,6 +435,7 @@ func (m *Matrix) GetMatrixRank(fieldOrder *big.Int) (uint64, error) {
 	if upper.numberRow < upper.numberColumn {
 		upper = upper.Transpose()
 	}
+
```
Comment (markya0616) [inline crypto/matrix/matrix.go:438]: rm this change

File: crypto/matrix/matrix.go (hunk 1 above)
Comment (cychuang0924) [reply crypto/matrix/matrix.go:531]: XD OK

File: crypto/matrix/matrix_test.go (hunk 2 above)
Comment (cychuang0924) [reply crypto/matrix/matrix_test.go:675]: OK

**Pull request merged**